learnhubmk/app
# Implement login authentication using NextAuth

Changes implemented:

- Integrated NextAuth with the credentials provider to handle user login via email and password.

- Implemented session management using JWT strategy with custom callbacks to store accessToken, userId, and other user information.

- Created custom authentication options in authOptions.ts to connect with our backend API for user verification.

- Extended NextAuth types to include custom fields like accessToken and userId.
- Updated the next-auth.d.ts file to handle custom session data and resolved TypeScript errors.
- Secured the content-panel/dashboard route using server-side checks with getServerSession. If a user is not authenticated, they are redirected to the /login page. (for testing purposes)
- Added server-side protection to ensure only authenticated users can access the dashboard.  (for testing purposes)

How to Test:

- Navigate to content-panel/login and attempt to log in using valid content panel credentials.

- Verify that authenticated users are redirected to content-panel/dashboard and can access protected content.
- Test the logout functionality to ensure it clears the session and redirects users back to  content-panel/login.
